composer.json

```diff
@@ -45,11 +45,14 @@
     },
     "minimum-stability": "dev",
     "prefer-stable": true,
+    "extra": {
+        "NOTE: symfony/dependency-injection": "symfony/dependency-injection is required for PHPStan to pass due to Rector prefixing."
+    },
```

Comment:
This is a bit unconventional, but there isn't a way to document in JSON. I wanted to highlight _why_ we have it as an extra `require-dev`. This wasn't caught before because the PHPStan branch wasn't merged.

Reply:
Since we removed Rector prefixing, is this still true?

`symfony/dependency-injection is required for PHPStan to pass due to Rector prefixing.`

Reply:
Turns out, yes, it is true. We can't add `rector/rector-src` which would work around this _without_ requiring PHP 8.0 _for development_. So our `require-dev` would look like this:

```
    "require-dev": {
        "php": "^8.0",
        "phpstan/extension-installer": "^1.1",
        "phpstan/phpstan": "^0.12.82",
        "phpstan/phpstan-deprecation-rules": "^0.12.6",
        "rector/rector-src":  "~0.11.0",
        "symfony/yaml": "^5"
    }
```

This wouldn't effect anyone _running_ Rector on their Drupal sites. Just contributing to the problem.

Reply:
After reviewing the development sandbox environment, I do not think there would be a problem either.

Reply:
I'm still not sure I grok the full extent of this change. This `require-dev` would go into drupal-rector, so wouldn't it fail to install on PHP 7?

I think I just need testing instructions with the revised PR.

Reply:
Let me make the PR updates and instructions